location-service/src/main/java/com/movie/recommender/location/controller/LocationController.java

```diff
@@ -0,0 +1,50 @@
+package com.movie.recommender.location.controller;
+
+import com.movie.recommender.location.service.LocationService;
+import com.movie.recommender.location.dto.LocationsDTO;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.HttpStatus;
+import lombok.extern.slf4j.Slf4j;
+import jakarta.validation.Valid;
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+@Slf4j
+@RestController
+@RequestMapping("api/v1/locations")
+public class LocationController {
+
+    private final LocationService locationService;
+
+    public LocationController(LocationService locationService) {
+        this.locationService = locationService;
+    }
+
+    @PostMapping("/")
+    public ResponseEntity<?> createLocation(@RequestBody @Valid LocationsDTO locationDTO) {
+        log.info("Creating location with: latitude: {}, longitude: {}, timestamp: {}, user_id {}",
```

Comment:
simplify it by just passing the object and using a custom toString
`log.info("Creating location: {}", locationDTO);`